### PR TITLE
Downshift integration

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

**Comment by ShahAnuj2610** [comment]
Quick working demo:

downshift_demo

File: src/components/VueSearchbox.jsx [hunk 1]
```diff
@@ -44,7 +48,21 @@ const VueSearchbox = {
...
+      currentValue,
+      suggestionsList: defaultSuggestions || [],
+      isOpen: false,
+      error: null,
+      loading: false
+    };
+    return this.state;
+  },
   created() {
+    console.log(this.$props);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:65]: Let's remove this now.

File: src/components/VueSearchbox.jsx [hunk 2]
```diff
@@ -67,7 +85,7 @@ const VueSearchbox = {
 
       try {
         const transformQuery = query => {
-          if (defaultQuery) return defaultQuery(query, this.state.currentValue);
+          if (defaultQuery) return defaultQuery(query, this.$data.currentValue);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:88]: Did you try using this?

File: src/components/VueSearchbox.jsx [hunk 3]
```diff
@@ -86,14 +104,209 @@ const VueSearchbox = {
...
+        this.searchBase.subscribeToStateChanges(this.setStateValue, [
+          "suggestions"
+        ]);
+
+        this.searchBase.onValueChange = nextValue => {
+          this.currentValue = nextValue;
+          this.$emit("valueChange", nextValue);
+        };
       } catch (e) {
         console.error(e);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:116]: We don't have `renderError` or `onError` over here ?

File: src/components/VueSearchbox.jsx [hunk 4]
```diff
@@ -86,14 +104,209 @@ const VueSearchbox = {
...
       }
+    },
+    setStateValue({ suggestions = {} }) {
+      this.suggestionsList = (suggestions.next && suggestions.next.data) || [];
+    },
+    onInputChange(event) {
+      console.log("inside oninputchange", event);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:123]: Console.log here as well

File: src/components/VueSearchbox.jsx [hunk 5]
```diff
@@ -86,14 +104,209 @@ const VueSearchbox = {
...
+    setValue({ value, isOpen = true }) {
+      console.log("inside setValue", this.searchBase);
+      const { debounce } = this.$props;
+      this.isOpen = isOpen;
+      if (debounce > 0)
+        this.searchBase.setValue(value, { triggerQuery: false });
+      this.triggerSuggestionsQuery(value);
+    },
+    triggerSuggestionsQuery(value) {
+      console.log("inside triggerSuggestionsQuery", value);
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:142]: Let's remove all console statements. You can find using search and remove them all.

File: src/components/VueSearchbox.jsx [hunk 6]
```diff
@@ -86,14 +104,209 @@ const VueSearchbox = {
...
+                      >
+                        <SuggestionItem
+                          currentValue={currentValue}
+                          suggestion={item}
+                        />
+                      </li>
+                    ))}
+                  </ul>
+                ) : (
+                  <div>no suggestion</div>
```
Comment (jyash97) [inline src/components/VueSearchbox.jsx:302]: Should be `No Suggestions` or what we have on RS

File: src/components/VueSearchbox.jsx (hunk 3 above)
Comment (ShahAnuj2610) [reply src/components/VueSearchbox.jsx:116]: it's in the subsequent PR

File: src/components/VueSearchbox.jsx (hunk 6 above)
Comment (ShahAnuj2610) [reply src/components/VueSearchbox.jsx:302]: subsequent PR

**Pull request merged**